src/resolver.cpp

```diff
@@ -235,211 +222,95 @@ std::string UsdOpenAssetIOResolver::_CreateIdentifier(
           // resolve to an absolute path, making the anchorAssetPath redundant
           // (for now).
           // TODO(DF): Allow the manager to provide an identifier representing
-          //  an "anchored" entity reference.
+          //  an "anchored" entity reference via getWithRelationship.
           identifier = assetPath;
         } else {
-          identifier = ArDefaultResolver::_CreateIdentifier(
-              assetPath,
-              locationInManagerContextForEntity(manager_, readContext_, anchorAssetPath));
+          identifier = ArDefaultResolver::_CreateIdentifier(assetPath, anchorAssetPath);
```

Comment:
[question] So is `anchorAssetPath` always a resolved file path at this point?

Reply:
That is my understanding, as it's a `ArResolvedPath`, and I saw in debug logs that there is a nested resolve call in `_CreateIdentifier`. We used to have to resolve it manually as we weren't resolving in `_Resolve`.

Reply:
I can double check though, as we have a test data set where the parent is a BAL entity.

Reply:
Double checked, it always seems to be resolved.